fix regression in shard checkpoint loading in AutoTP Path caused by qkv_copy() is deleted and add UT case for shard checkpoint loading in AutoTP

sywangyi opened the pull request:
1) I add UT for the shard loading in AutoTP path, because I find the code could not be tested in CI and error like "qkv_copy() is replaced by strided_copy()" is not found when merge.
2) the UT covers "bigscience/bloom-560m", "EleutherAI/gpt-j-6B", "EleutherAI/gpt-neo-125M", "facebook/opt-125m". and I also fix the problem found in gpt-neo-125m and opt-125m

sywangyi commented:
@tjruwase @delock @yao-matrix

tjruwase commented on deepspeed/module_inject/replace_module.py:
```diff
@@ -474,9 +475,10 @@ def update_mp_params(child):
+        def _replace_module(r_module, prev_name='', prev_class_name=''):
             for name, child in r_module.named_children():
-                checking_key = prefix + '.' + prev_name + '.' + name + '.' if prev_name != "" else prefix + '.' + name + '.'
+                class_name = prev_name if prev_class_name == "" else prev_class_name + '.' + prev_name if prev_name != "" else prev_class_name
```
Please simplify with regular `if ... elif ...else`

sywangyi commented:
@tjruwase I added another commit for https://github.com/microsoft/DeepSpeed/commit/db26f8b41325be2a7f7af8b386b4e8951a5a76c9, the latest merged code suppose only KI path support shard loading, actually I have already added the support, see the usage in UT
